time-tracker/src/reducers.jsx

```diff
@@ -0,0 +1,61 @@
+import { createAction, createReducer } from "@reduxjs/toolkit";
+
+export const changeCurrentTask = createAction("changeCurrentTask");
+export const pauseAllTasks = createAction("pauseAllTasks");
+export const addNewTask = createAction(
+  "addNewTask",
+  (taskName = "TASK NAME NOT SET") => {
+    return {
+      payload: {
+        taskName: taskName,
+      },
+    };
+  },
+);
+export const runOncePerSecond = createAction("runOncePerSecond");
+
+const initialState = {
+  tasks: {},
+  totalTimePassedInSeconds: 0,
+  currentActiveTask: null,
+};
+
+export const reducer = createReducer(initialState, (builder) => {
```

Comment:
Reducers in redux can be composed, so the main reducer is usually called rootReducer